### ISSUE TITLE
Error during installation: relocation R_X86_64_32 against

### ISSUE THREAD

**Issue by yangmaoshan** [GalacticDynamics-Oxford/Agama#7]
I installed python 2.7.18 in my local directory /software/jwu549/myPy2/ on a HPC cluster using the source code downloaded from python.org. (I did not use the default python on the cluster because the corresponding python-dev was not installed. )To install agama, I used the following command:
`/software/jwu549/myPy2/bin/python -m pip install agama`

I got many messages like this:  
relocation R_X86_64_32 against `.rodata.str1.8'  can not be used when making a shared object; recompile with -fPIC

Below is a screenshot of the message output
![image](https://user-images.githubusercontent.com/34459209/103820839-a0b90900-5032-11eb-9359-8a02346c3235.png)

The same error occurred when trying to install agama from the source code using setup.py.

**Comment by eugvas** [comment]
So you're compiling your own Python version, then installing Agama, right?
I've never seen a need to do this - have you checked that the installation does not work with the system-wide Python? I think it does need some header files, indeed, but I'd be surprised if they were not available on an HPC system.
If you really wish to follow your current approach though, then you have full control on how your version of Python is built, so need to do one of the following:
- either add a -fPIC file during the compilation of the static library libpython2.7.a
- or build a dynamic library libpython2.7.so (or whatever way it is called).
The installation script in Agama tries all these possible link options (static or dynamic), and at least one of them should work (but as it rightly points out, for the static library to be correctly linked, it has to be compiled with the -fPIC option).

**Comment by yangmaoshan** [comment]
Thank you for the quick reply! I abandoned my approach. I contacted the administrators of the HPC cluster. They installed python-dev and then Agama got installed successfully using the system-wide Python.

For people who want to install Agama on top of your own python version, I think this problem can be solved by using 
`./configure --enable-shared`
in the process of building and installing your own Python. More information related to this issue can be found here [http://blog.dscpl.com.au/2015/06/installing-custom-python-version-into.html](url)
